Dynamic error sampling Python

szokeasaurusrex opened the pull request:
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Document the Python `error-sampler` option we introduced in getsentry/sentry-python#2456. Closes #8305.


## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

sentrivana commented on src/platforms/common/configuration/sampling.mdx:
```diff
@@ -41,6 +41,30 @@ Changing the error sample rate requires re-deployment. In addition, setting an S
+
+### Dynamically Sampling Error Events
+
+It is also possible to sample error events dymanically, by setting the <PlatformIdentifier name="error-sampler" /> to a function that returns the desired sample rate for the event. The <PlatformIdentifier name="error-sampler" /> receives two arguments, the <PlatformIdentifier name="event" /> and the <PlatformIdentifier name="hint" />, when called, which it can inspect in order to inform the sampl ...
```
```suggestion
It is also possible to sample error events dynamically, by setting the <PlatformIdentifier name="error-sampler" /> to a function that returns the desired sample rate for the event. The <PlatformIdentifier name="error-sampler" /> receives two arguments, the <PlatformIdentifier name="event" /> and the <PlatformIdentifier name="hint" />, when called, which it can inspect in order to inform the sampling decision.
```

sentrivana reviewed (approved):
Please also add `error_sampler` to the [Basic Options](https://docs.sentry.io/platforms/python/configuration/options/) (best somewhere close to `sample_rate`)

shanamatthews commented on src/platforms/common/configuration/sampling.mdx:
```diff
@@ -41,6 +41,30 @@ Changing the error sample rate requires re-deployment. In addition, setting an S
+
+<Alert level="warning">
+
+The <PlatformIdentifier name="error-sampler" /> function is responsible for returning a valid value. The only valid values are **floating-point numbers** between `0.0` and `1.0`, inclusive (to indicate the probability that an error gets sampled), and **booleans** (to indicate whether or not to sample the error).
```
```suggestion
Your <PlatformIdentifier name="error-sampler" /> function must return a valid value. A valid value is a **floating-point number** between `0.0` and `1.0` (inclusive) indicating the probability an error gets sampled) or a **boolean** indicating whether or not to sample the error.
```

shanamatthews reviewed (approved):
Looks good! Just some suggestions to clarify/simplify the language.

szokeasaurusrex commented:
I believe I have addressed the concerns from the previous reviews; I would appreciate a quick re-review focusing on the changes I made since the last review

sentrivana reviewed (approved):
👍🏻

shanamatthews reviewed (approved):
Looks good to me! Thanks for making those improvements!
